controllers/capabilities/developeruser_controller.go

```diff
@@ -74,12 +80,67 @@ func (r *DeveloperUserReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 		reqLogger.V(1).Info(string(jsonData))
 	}
 
+	// DeveloperUser has been marked for deletion
+	if developerUserCR.GetDeletionTimestamp() != nil && controllerutil.ContainsFinalizer(developerUserCR, developerUserFinalizer) {
+		err = r.removeDeveloperUserFrom3scale(developerUserCR)
+		if err != nil {
+			r.EventRecorder().Eventf(developerUserCR, corev1.EventTypeWarning, "Failed to delete developer user", "%v", err)
+
+			// Update status with err
+			statusResult, statusUpdateErr := NewDeveloperUserStatusReconciler(r.BaseReconciler, developerUserCR, nil, "", nil, err).Reconcile()
+			if statusUpdateErr != nil {
+				return ctrl.Result{}, fmt.Errorf("Failed to update developers user status: %w", statusUpdateErr)
+			}
+
+			if statusResult.Requeue {
+				return statusResult, nil
+			}
+
+			return ctrl.Result{}, err
```

Comment:
![Screenshot from 2022-05-31 14-59-02](https://user-images.githubusercontent.com/24636860/171191788-822135f8-c47d-4016-9c63-e9c3005dc064.png)

This will add the last admin error to the CR condition field